randovania/generator/pickup_pool/pool_creator.py

```diff
@@ -55,43 +55,43 @@ def calculate_pool_results(layout_configuration: BaseConfiguration,
     return base_results
 
 
-def calculate_pool_pickup_count(layout: BaseConfiguration) -> tuple[int, int]:
+def calculate_pool_pickup_count(layout: BaseConfiguration) -> dict[LocationCategory | str, tuple[int, int]]:
     """
     Calculate how many pickups are needed for given layout, with how many spots are there.
     :param layout:
     :return:
     """
     game_description = filtered_database.game_description_for_layout(layout)
 
-    num_pickup_nodes = game_description.world_list.num_pickup_nodes
+    result = {
+        cat: [0, 0]
+        for cat in LocationCategory
+    }
+
+    for node in game_description.world_list.iterate_nodes():
+        if isinstance(node, PickupNode):
+            result[node.location_category][1] += 1
+
     pool_results = calculate_pool_results(layout, game_description,
                                           rng_required=False)
-    min_starting_pickups = layout.standard_pickup_configuration.minimum_random_starting_pickups
+    result["Starting"] = [0, layout.standard_pickup_configuration.minimum_random_starting_pickups]
```

Comment:
curious why we're assigning the number of nodes here, rather than number of pickups

Reply:
Minimum random starting pickups means there are more "places" that pickups can be placed at, so the pool can have more pickups in total. That's the same as having extra nodes.